devimint/src/federation.rs

```diff
@@ -142,6 +142,47 @@ impl Client {
         Ok(())
     }
 
+    /// Client to join a federation
+    pub async fn restore_federation(&self, invite_code: String, mnemonic: String) -> Result<()> {
```

Comment:
[nit] I believe this is unused. If we want to keep it, it might worth updating the docs and the debug log.